iceoryx_binding_c/test/moduletests/test_service_discovery.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/runtime/service_discovery.hpp"
+#include "iceoryx_posh/testing/roudi_gtest.hpp"
+
+using namespace iox;
+using namespace iox::runtime;
+
+extern "C" {
+#include "iceoryx_binding_c/publisher.h"
+#include "iceoryx_binding_c/runtime.h"
+#include "iceoryx_binding_c/service_discovery.h"
+}
+
+namespace
+{
+using namespace ::testing;
+using description_vector = cxx::vector<iox_service_description_t, MAX_FINDSERVICE_RESULT_SIZE>;
+
+class iox_service_discovery_test : public RouDi_GTest
+{
+  public:
+    void SetUp() override
+    {
+        iox_runtime_init("runtime");
```

Comment:
Shouldn't we use the runtime mock here. Why does this work I wonder?!

Reply:
Because the actual runtime creates a publisher port and the sut (ServiceDiscovery) receives service registry data from it as there is a Roudi Environment active to connect them. It is fine like this, we do not need a mock here.